### PR TITLE
Add support for Eth mappings

### PR DESCRIPTION
**Pull request by elmattic**
## Summary of changes

Changes introduced in this pull request:

- Introduce a new ParityDB namespace for storing Ethereum mappings such as `Hash` to `TipsetKey` and `Hash` to message `Cid`.
  This is required by methods like:
  - EthGetBlockByHash
  - EthGetBlockTransactionCountByHash
  - EthGetMessageCidByTransactionHash
- Increment Forest version to `0.19.0`

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by aatifsyed — APPROVED** [review]

File: src/daemon/db_util.rs [hunk 1]
```diff
@@ -142,6 +149,96 @@ async fn transcode_into_forest_car(from: &Path, to: &Path) -> anyhow::Result<()>
     Ok(())
 }
 
+pub fn populate_eth_mappings<DB>(
```
Comment (lemmih) [inline src/daemon/db_util.rs:152]: Could you describe what we're mapping and how many mappings we expect in a comment here?

File: src/daemon/db_util.rs [hunk 2]
```diff
@@ -142,6 +149,96 @@ async fn transcode_into_forest_car(from: &Path, to: &Path) -> anyhow::Result<()>
...
+            }
+        }
+        state_manager.chain_store().put_tipset_key(ts.key())?;
+    }
+    let _ = process_signed_messages(state_manager, &delegated_messages)?;
+
+    Ok(())
+}
+
+fn process_signed_messages<DB>(
```
Comment (lemmih) [inline src/daemon/db_util.rs:188]: How are the signed messages being processed? Describe in a comment.

File: src/db/migration/migration_map.rs [hunk 3]
```diff
@@ -86,20 +89,26 @@ pub struct Migration {
...
-        let migrated_db = self.migrator.migrate(chain_data_path)?;
+        let migrated_db = self.migrator.migrate(chain_data_path, config)?;
         self.post_checks(chain_data_path)?;
 
         let new_db = chain_data_path.join(format!("{}", self.to));
+        info!(
+            "Renaming database {} to {}",
+            migrated_db.display(),
+            new_db.display()
+        );
```
Comment (lemmih) [inline src/db/migration/migration_map.rs:107]: Is this important information?

File: src/db/parity_db.rs [hunk 4]
```diff
@@ -69,6 +75,13 @@ impl DbColumn {
                         compression,
                         ..Default::default()
                     },
+                    DbColumn::EthMappings => parity_db::ColumnOptions {
+                        preimage: false,
+                        // This is needed for key retrieval.
+                        btree_index: true,
```
Comment (lemmih) [inline src/db/parity_db.rs:81]: Why is this needed for key retrieval?

**Review by lemmih — CHANGES_REQUESTED** [review]

File: src/daemon/db_util.rs (hunk 1 above)
Comment (elmattic) [reply src/daemon/db_util.rs:152]: Sure.

File: src/db/parity_db.rs (hunk 4 above)
Comment (elmattic) [reply src/db/parity_db.rs]: I'm not 100% sure. Hash column indexes should work at the moment (I'm testing them). But what if we need to implement GC on `Hash` to message the `Cid` later (like Lotus is doing, see https://github.com/filecoin-project/lotus/blob/master/node/impl/full/txhashmanager.go#L122).

Comment (elmattic) [reply src/db/parity_db.rs]: Since we want iteration for later implementation of GC on Eth mappings, let's keep it set to `true`.

Comment (ruseinov) [reply src/db/parity_db.rs]: Actually we don't need it for the GC. But we need it to retrieve data by CID when exposing a set of RPC endpoints.

**Comment by ruseinov** [comment]
Some things to consider: https://github.com/ChainSafe/forest/blob/6f60fedc52a10d8575a6f06950fb278476dedbb0/src/db/parity_db.rs#L126 needs to account for the new column as well as all the places this method is being called.

File: src/db/parity_db.rs (hunk 4 above)
Comment (elmattic) [reply src/db/parity_db.rs]: Hash indexes are enough for iterating on column values. But I'm unsure if we can retrieve the key for the value for the mapping `{Hash => message Cid}.`

Comment (elmattic) [reply src/db/parity_db.rs]: I was able to retrieve the transaction `Hash` from the mapping value (message `Cid` and timestamp pair).

**Review by lemmih — APPROVED** [review]

**Review by ruseinov — APPROVED** [review]
